api/projects/index.ts

```diff
@@ -10,7 +11,8 @@ export class SmartlingProjectsApi extends SmartlingBaseApi {
         this.entrypoint = `${smartlingApiBaseUrl}/projects-api/v2/projects`;
     }
 
-    async getProjectDetails(projectId: string): Promise<ProjectDto> {
+    @Validate
+    async getProjectDetails(@IsNotEmpty projectId: string): Promise<ProjectDto> {
```

Comment:
@dimitrystd it's just an example of how it can be used (SUBM api class is in private sdk)

Reply:
Looks interesting. Just two concerns:
- It only works with simple types. If we pass an object of `Parameters` class then we can't validate it. Or need to write more code. 
  Using `!` and strict compilation can give us similar results.
- We implement just another validation package.

Reply:
We would need to refactor here and there to support `strictNullChecks` mode.